operator/initc/internal/wait_test.go

```diff
@@ -0,0 +1,963 @@
+				var targetClique string
+				for cliqueName := range tt.initialDeps {
+					if strings.HasPrefix(tt.pod.Name, cliqueName) {
+						targetClique = cliqueName
+						break
+					}
+				}
+
+				require.NotEmpty(t, targetClique, "Pod should belong to a tracked PodClique")
+				assert.Equal(t, tt.expectedReadyCount, state.currentPCLQReadyPods[targetClique].Len())
+			}
+		})
+	}
+}
+
+// TestCheckAllParentsReady tests the logic for determining when all parent PodCliques are ready.
+// It validates different scenarios of ready pod counts vs minimum requirements.
+func TestCheckAllParentsReady(t *testing.T) {
+	tests := []struct {
+		// PodClique dependencies with minimum required replicas
+		dependencies map[string]int
+		// Current ready pod counts for each PodClique
+		readyCounts map[string]int
+		// Whether all parents should be considered ready
+		expectedReady bool
+	}{
+		// All PodCliques meet minimum requirements exactly
+		{
+			dependencies:  map[string]int{"clique-a": 2, "clique-b": 3},
+			readyCounts:   map[string]int{"clique-a": 2, "clique-b": 3},
+			expectedReady: true,
+		},
+		// All PodCliques exceed minimum requirements
+		{
+			dependencies:  map[string]int{"clique-a": 2, "clique-b": 3},
+			readyCounts:   map[string]int{"clique-a": 5, "clique-b": 4},
+			expectedReady: true,
+		},
+		// One PodClique below minimum requirement
+		{
+			dependencies:  map[string]int{"clique-a": 2, "clique-b": 3},
+			readyCounts:   map[string]int{"clique-a": 2, "clique-b": 2},
+			expectedReady: false,
+		},
+		// All PodCliques below minimum requirements
+		{
+			dependencies:  map[string]int{"clique-a": 2, "clique-b": 3},
+			readyCounts:   map[string]int{"clique-a": 1, "clique-b": 1},
+			expectedReady: false,
+		},
+		// Single PodClique meets requirement
+		{
+			dependencies:  map[string]int{"single-clique": 1},
+			readyCounts:   map[string]int{"single-clique": 1},
+			expectedReady: true,
+		},
+		// Single PodClique below requirement
+		{
+			dependencies:  map[string]int{"single-clique": 3},
+			readyCounts:   map[string]int{"single-clique": 2},
+			expectedReady: false,
+		},
+		// Zero requirements should always be ready
+		{
+			dependencies:  map[string]int{"zero-clique": 0},
+			readyCounts:   map[string]int{"zero-clique": 0},
+			expectedReady: true,
+		},
+		// Empty dependencies should be ready
+		{
+			dependencies:  map[string]int{},
+			readyCounts:   map[string]int{},
+			expectedReady: true,
+		},
+	}
+
+	for i, tt := range tests {
+		t.Run(fmt.Sprintf("case_%d", i), func(t *testing.T) {
+			// Create state with test dependencies
+			state := &ParentPodCliqueDependencies{
+				pclqFQNToMinAvailable: tt.dependencies,
+				currentPCLQReadyPods:  make(map[string]sets.Set[string]),
+			}
+
+			// Set up ready pod counts
+			for cliqueName, readyCount := range tt.readyCounts {
+				podSet := sets.New[string]()
+				// Add dummy pod names to reach the desired count
+				for i := 0; i < readyCount; i++ {
```

Comment:
Range loops are simpler and are the convention for [0, n) iterations.

```suggestion
				for i := range readyCount {
```